### PR TITLE
`Stepper` - Convert components to TypeScript

### PR DESCRIPTION
**Pull request by Dhaulagiri**
### :pushpin: Summary

Converts the Stepper components to TS

### :link: External links

Jira ticket: [HDS-2712](https://hashicorp.atlassian.net/browse/HDS-2712)

***

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2712]: https://hashicorp.atlassian.net/browse/HDS-2712?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/template-registry.ts [hunk 1]
```diff
@@ -101,6 +103,15 @@ export default interface HdsComponentsRegistry {
   'hds/reveal/toggle/button': typeof HdsRevealToggleButtonComponent;
   HdsRevealToggleButtonComponent: typeof HdsRevealToggleButtonComponent;
 
+  // Stepper
+  'Hds::Stepper::Step::Indicator': typeof HdsStepperIndicatorStepComponent;
+  'hds/stepper/step/indicator': typeof HdsStepperIndicatorStepComponent;
+  HdsStepperStepIndicator: typeof HdsStepperIndicatorStepComponent;
```
Comment (Dhaulagiri) [inline packages/components/src/template-registry.ts:109]: I got kinda dizzy trying to ensure I got `Stepper` `Step` & `Indicator` In the right order here. I _think_ it's correct but a second set of eyes would be helpful. Ditto the section below as well.

Comment (didoo) [inline packages/components/src/template-registry.ts:109]:
```suggestion
  'Hds::Stepper::Step::Indicator': typeof HdsStepperStepIndicatorComponent;
  'hds/stepper/step/indicator': typeof HdsStepperStepIndicatorComponent;
  HdsStepperStepIndicator: typeof HdsStepperStepIndicatorComponent;
```

File: packages/components/src/template-registry.ts [hunk 2]
```diff
@@ -101,6 +103,15 @@ export default interface HdsComponentsRegistry {
...
+
+  'Hds::Stepper::Task::Indicator': typeof HdsStepperIndicatorTaskComponent;
+  'hds/stepper/task/indicator': typeof HdsStepperIndicatorTaskComponent;
+  HdsStepperTaskIndicator: typeof HdsStepperIndicatorTaskComponent;
```
Comment (didoo) [inline packages/components/src/template-registry.ts:113]:
```suggestion
  'Hds::Stepper::Task::Indicator': typeof HdsStepperTaskIndicatorComponent;
  'hds/stepper/task/indicator': typeof HdsStepperTaskIndicatorComponent;
  HdsStepperTaskIndicator: typeof HdsStepperTaskIndicatorComponent;
```

File: packages/components/src/template-registry.ts [hunk 3]
```diff
@@ -18,6 +18,8 @@ import type HdsLinkInlineComponent from './components/hds/link/inline';
 import type HdsLinkStandaloneComponent from './components/hds/link/standalone';
 import type HdsRevealComponent from './components/hds/reveal';
 import type HdsRevealToggleButtonComponent from './components/hds/reveal/toggle/button';
+import type HdsStepperIndicatorStepComponent from './components/hds/stepper/step/indicator';
+import type HdsStepperIndicatorTaskComponent from './components/hds/stepper/task/indicator';
```
Comment (didoo) [inline packages/components/src/template-registry.ts:22]:
```suggestion
import type HdsStepperStepIndicatorComponent from './components/hds/stepper/step/indicator';
import type HdsStepperTaskIndicatorComponent from './components/hds/stepper/task/indicator';
```

File: packages/components/src/components/hds/stepper/task/indicator.ts [hunk 4]
```diff
@@ -6,24 +6,34 @@
...
+
+interface HdsStepperIndicatorTaskSignature {
+  Args: {
+    isInteractive?: boolean;
+    status: HdsStepperStatuses;
+  };
+  Element: HTMLDivElement;
+}
+
+export default class HdsStepperIndicatorTaskComponent extends Component<HdsStepperIndicatorTaskSignature> {
```
Comment (didoo) [inline packages/components/src/components/hds/stepper/task/indicator.ts:28]:
```suggestion
export default class HdsStepperTaskIndicatorComponent extends Component<HdsStepperTaskIndicatorSignature> {
```

File: packages/components/src/components/hds/stepper/task/indicator.ts [hunk 5]
```diff
@@ -6,24 +6,34 @@
...
+  HdsStepperStatusToIconsValues,
+} from '../types.ts';
+import type { HdsStepperStatuses } from '../types.ts';
+
+export const DEFAULT_STATUS = HdsStepperStatusesValues.Incomplete;
+export const STATUSES: string[] = Object.values(HdsStepperStatusesValues);
+
+export const MAPPING_STATUS_TO_ICONS = HdsStepperStatusToIconsValues;
+
+interface HdsStepperIndicatorTaskSignature {
```
Comment (didoo) [inline packages/components/src/components/hds/stepper/task/indicator.ts:20]:
```suggestion
interface HdsStepperTaskIndicatorSignature {
```

File: packages/components/src/components/hds/stepper/step/indicator.ts [hunk 6]
```diff
@@ -6,18 +6,30 @@
...
+import type { HdsStepperStatuses } from '../types.ts';
 
-export default class HdsStepperIndicatorStepIndexComponent extends Component {
+export const DEFAULT_STATUS = HdsStepperStatusesValues.Incomplete;
+export const STATUSES: string[] = Object.values(HdsStepperStatusesValues);
+
+interface HdsStepperIndicatorStepSignature {
+  Args: {
+    isInteractive?: boolean;
+    status: HdsStepperStatuses;
```
Comment (didoo) [inline packages/components/src/components/hds/stepper/step/indicator.ts:18]: See: https://helios.hashicorp.design/components/stepper?tab=code#stepperstepindicator
```suggestion
    status: HdsStepperStatuses;
    isInteractive?: boolean;
```

File: packages/components/src/components/hds/stepper/step/indicator.ts [hunk 7]
```diff
@@ -6,18 +6,30 @@
...
-export const DEFAULT_STATUS = 'incomplete';
-export const STATUSES = ['incomplete', 'progress', 'processing', 'complete'];
+import { HdsStepperStatusesValues } from '../types.ts';
+import type { HdsStepperStatuses } from '../types.ts';
 
-export default class HdsStepperIndicatorStepIndexComponent extends Component {
+export const DEFAULT_STATUS = HdsStepperStatusesValues.Incomplete;
+export const STATUSES: string[] = Object.values(HdsStepperStatusesValues);
+
+interface HdsStepperIndicatorStepSignature {
```
Comment (didoo) [inline packages/components/src/components/hds/stepper/step/indicator.ts:15]:
```suggestion
interface HdsStepperStepIndicatorSignature {
```

File: packages/components/src/components/hds/stepper/step/indicator.ts [hunk 8]
```diff
@@ -6,18 +6,30 @@
...
+  Args: {
+    isInteractive?: boolean;
+    status: HdsStepperStatuses;
+    text: string;
+  };
+  Element: HTMLDivElement;
+}
+
+export default class HdsStepperIndicatorStepComponent extends Component<HdsStepperIndicatorStepSignature> {
```
Comment (didoo) [inline packages/components/src/components/hds/stepper/step/indicator.ts:24]:
```suggestion
export default class HdsStepperStepIndicatorComponent extends Component<HdsStepperStepIndicatorSignature> {
```

File: packages/components/src/components/hds/stepper/task/indicator.ts [hunk 9]
```diff
@@ -6,24 +6,34 @@
...
+  Args: {
+    isInteractive?: boolean;
+    status: HdsStepperStatuses;
```
Comment (didoo) [inline packages/components/src/components/hds/stepper/task/indicator.ts:23]: See: https://helios.hashicorp.design/components/stepper?tab=code#steppertaskindicator
```suggestion
    status: HdsStepperStatuses;
    isInteractive?: boolean;
```

**Review by didoo — COMMENTED** [review]
Left a few comments/suggestions

File: .changeset/purple-pears-hang.md [hunk 10]
```diff
@@ -0,0 +1,6 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+`Stepper::Step::Indicator` - Convert to TypeScript
+`Stepper::Task::Indicator` - Convert to TypeScript
```
Comment (alex-ju) [inline .changeset/purple-pears-hang.md:6]:
```suggestion
`Stepper::Step::Indicator` - Converted to TypeScript
`Stepper::Task::Indicator` - Converted to TypeScript
```

**Review by didoo — APPROVED** [review]
👍

**Pull request merged**